ls1intum/Artemis
# Show reachable scores in course statistics and fix tooltips of scores

# Checklist

### Motivation and Context
Course statistics show (relative) scores even though there are still pending results.

### Description
Statistics now show the absolute points, reachable points (where pending results are not included), and total points. Reachable and total include their respective relative scores. The total diagram on the left now also doesn't include exercises that haven't been assessed yet.

### Steps for Testing


1. Log in to Artemis
2. Navigate to a Course Overview (prefereably one where the user has results for exercises)
3. Go to Statistics
4. Check whether Your points, Reachable points and Total points display the correct numbers.
-> in reachable included should be every task where there is a finished result, so assessed submissions and exercises not participated in.
4.1 Check if points are correctly displayed on the tooltip of the bar chart.
5. Add new exercises with fitting due dates and assessment due dates.
6. Participate in those exercises and assess them (or not).
7. Repeat step 4.

Tooltips of programming exercises were the main source of inconsistencies. Take an extra look if the tooltips are correct for those.

### Screenshots


Before:
![Screenshot_2](https://user-images.githubusercontent.com/57408107/83627730-b398c600-a597-11ea-99f6-65a279815c88.png)
After:
![Screenshot_1](https://user-images.githubusercontent.com/57408107/83975399-140b6880-a8f4-11ea-9812-ef1c980fb124.png)
